src/sonic_ax_impl/mibs/ietf/rfc2737.py

```diff
@@ -1,32 +1,107 @@
+"""
+MIB implementation defined in RFC 2737
+"""
+
 from enum import Enum, unique
 from bisect import bisect_right
-import subprocess
 
 from swsssdk import SonicV2Connector
+from swsssdk import port_util
 from ax_interface import MIBMeta, MIBUpdater, ValueType, SubtreeMIBEntry
 
+from sonic_ax_impl import mibs
 
 @unique
 class PhysicalClass(int, Enum):
     """
     Physical classes defined in RFC 2737.
     """
-    OTHER = 1
-    UNKNOWN = 2
-    CHASSIS = 3
-    BACKPLANE = 4
-    CONTAINER = 5
+
+    OTHER       = 1
+    UNKNOWN     = 2
+    CHASSIS     = 3
+    BACKPLANE   = 4
+    CONTAINER   = 5
     POWERSUPPLY = 6
-    FAN = 7
-    SENSOR = 8
-    MODULE = 9
-    PORT = 10
-    STACK = 11
+    FAN         = 7
+    SENSOR      = 8
+    MODULE      = 9
+    PORT        = 10
+    STACK       = 11
+
+
+@unique
+class XcvrInfoDB(bytes, Enum):
+    """
+    Transceiver info keys
+    """
+
+    TYPE              = b"type"
+    HARDWARE_REVISION = b"hardwarerev"
+    SERIAL_NUMBER     = b"serialnum"
+    MANUFACTURE_NAME  = b"manufacturename"
+    MODEL_NAME        = b"modelname"
+
+
+# Map used to generate sensor description 
+SENSOR_NAME_MAP = {
+    "temperature" : "Temperature",
+    "voltage"     : "Voltage",
+    "rx1power"    : "RX Power",
+    "rx2power"    : "RX Power",
+    "rx3power"    : "RX Power",
+    "rx4power"    : "RX Power",
+    "tx1bias"     : "TX Bias",
+    "tx2bias"     : "TX Bias",
+    "tx3bias"     : "TX Bias",
+    "tx4bias"     : "TX Bias",
+}
+
+
+def get_transceiver_data(xcvr_info):
+    """
+    :param xcvr_info: transceiver info dict
+    :return: tuple (type, hw_version, mfg_name, model_name) of transceiver;
+    Empty string if field not in xcvr_info
+    """
+
+    return (xcvr_info.get(xcvr_field.value, b"").decode("utf-8")
+            for xcvr_field in XcvrInfoDB)
+
+
+def get_transceiver_description(sfp_type, if_alias):
+    """
+    :param sfp_type: SFP type of transceiver
+    :param if_alias: Port alias name
+    :return: Transceiver decsription
+    """
+
+    return "{} for {}".format(sfp_type, if_alias)
+
+def get_transceiver_sensor_description(sensor, if_alias):
+    """
+    :param sensor: sensor key name
+    :param if_alias: interface alias
+    :return: description string about sensor
+    """
+
+    # assume sensors that is per channel in transceiver port
+    # has digit equals to channel number in the sensor's key name in DB
+    digit = list(filter(lambda c: c.isdigit(), sensor))
+    if len(digit) == 1:
```

Comment:
>digit [](start = 11, length = 5)

Are you assuming sensor number is 1-digit? Better remove relax it to multiple digits.

Reply:
No, not a sensor number. Usually sensor name has a lane number in its name. So I think assuming lane number is 1 digit is ok, otherwise - it is something different